src/backend/drm/node/mod.rs

```diff
@@ -120,22 +115,22 @@ impl Display for DrmNode {
 }
 
 impl IntoRawFd for DrmNode {
-    fn into_raw_fd(mut self) -> RawFd {
-        self.fd.take().unwrap()
+    fn into_raw_fd(self) -> RawFd {
+        let fd = self.fd;
+        mem::forget(self);
```

Comment:
But doesn't that leak memory?

Reply:
Nope:
> Takes ownership and “forgets” about the value without running its destructor.
>
> Any resources the value manages, such as heap memory or a file handle, will linger forever in an unreachable state. > However, it does not guarantee that pointers to this memory will remain valid.

Reply:
Plus the semantics defined by IntoRawFd indicate the caller takes ownership of the open fd

Reply:
Yes for the Fd, that is correct, but we leak the memory of "self".

Reply:
Misunderstood, stack variables are of course still deallocated, just not "dropped"